### PR TITLE
ZD153122 - Flexible homedir umask permission

### PR DESCRIPTION
**Pull request by kramfs**
Add input for users to set homedir umask

### CODE REVIEW

File: rll/enable-managed-login.sh [hunk 1]
```diff
@@ -17,6 +17,12 @@
 #       - text:auto
 #       - text:enable
 #       - text:disable
+#   UMASK:
+#     Input Type: single
+#     Category: RightScale
+#     Description: Permission for creating users homedir. Default umask value is '0026' which means 7751 or rwxrwxr-xr--.
+#     Required: true
+#     Default: text:0026
```
Comment (douglaswth) [inline rll/enable-managed-login.sh]: We should keep the default as `0022` since that is what we had before.

File: rll/enable-managed-login.sh [hunk 2]
```diff
@@ -195,12 +201,15 @@ enable)
     sudo bash -c "umask 0337 && printf '# Members of the rightscale_sudo group may gain root privileges\n%%rightscale_sudo ALL=(ALL) SETENV:NOPASSWD:ALL\n' > /etc/sudoers.d/90-rightscale-sudo-users"
   fi
 
+  # Ask user input what should be the homedir permission. If no input is given, set default value (0026) which is eq to 7751
+  : "${UMASK:=0026}"
```
Comment (douglaswth) [inline rll/enable-managed-login.sh]: Same as above.

**Review by douglaswth — CHANGES_REQUESTED** [review]

File: rll/enable-managed-login.sh (hunk 2 above)
Comment (kramfs) [reply rll/enable-managed-login.sh]: PR updated, default umask back to `0022`

**Comment by kramfs** [comment]
@douglaswth can you review this?

**Review by douglaswth — APPROVED** [review]
This looks fine now, but please communicate with the Titanium team before merging.

File: rll/enable-managed-login.sh [hunk 3]
```diff
@@ -195,12 +201,15 @@ enable)
     sudo bash -c "umask 0337 && printf '# Members of the rightscale_sudo group may gain root privileges\n%%rightscale_sudo ALL=(ALL) SETENV:NOPASSWD:ALL\n' > /etc/sudoers.d/90-rightscale-sudo-users"
   fi
 
+  # Ask user input what should be the homedir permission. If no input is given, set default value (0066) which is eq to 7755
```
Comment (douglaswth) [inline rll/enable-managed-login.sh]:
```suggestion
  # Ask user input what should be the homedir permission. If no input is given, set default value (0022) which is eq to 7755
```

File: rll/enable-managed-login.sh [hunk 4]
```diff
@@ -17,6 +17,12 @@
 #       - text:auto
 #       - text:enable
 #       - text:disable
+#   UMASK:
+#     Input Type: single
+#     Category: RightScale
+#     Description: Permission for creating users homedir. Default umask value is '0022' which means 7755 or rwxrwxr-xr-x.
```
Comment (douglaswth) [inline rll/enable-managed-login.sh]:
```suggestion
#     Description: Permission for creating users homedir. Default umask value is '0022' which means 0755 or rwxr-xr-x.
```

**Review by douglaswth — CHANGES_REQUESTED** [review]
A couple comment changes needed. Also, this will not change the configured umask if the script has been run previously with a different umask value.

File: rll/enable-managed-login.sh (hunk 3 above)
Comment (lisabien) [reply rll/enable-managed-login.sh]: Also, as in the first suggestion above:
https://github.com/rightscale/rightlink_scripts/pull/124/files#r351433568
`7755 or rwxrwxr-xr-x.` -> `0755 or rwxr-xr-x.`

**Review by lisabien — COMMENTED** [review]
LGTM with the suggested fixes.

**Pull request merged**